### DISCUSSION TITLE
BRouter position not mapped

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#205] [Q&A] [answered]
From position not mapped in existing datafile.
(New download from today 6/6/23)

What is wrong ? 

By BRouter offline = false
By BRouter online = fine.

[What is wrong.gpx.txt](https://github.com/devemux86/cruiser/files/11662349/What.is.wrong.gpx.txt)
![image](https://github.com/devemux86/cruiser/assets/54167201/8155d99d-3f07-4714-82f5-aa136c99dbf6)
pref: [What is wrong.zip](https://github.com/devemux86/cruiser/files/11662444/What.is.wrong.zip)
Brouter profile
Internal profile: car-cruiser

**Comment by devemux86** [comment]
@0709wiwiwi please provide more details so we can understand and reproduce the report.

- Cruiser profile
- BRouter settings
- Route example

**Comment by devemux86** [comment]
What do you have as "Waypoints search range" in BRouter (offline) advanced settings?
If we set it to 1000 m then this example works.

However it does not work when we move the start on the above highway=trunk:
https://www.openstreetmap.org/way/754690414

The bike profile always works, could it be something in car profiles that prevents it?

**Reply by 0709wiwiwi** [reply]
What do you have as "Waypoints search range" in BRouter (offline) advanced settings?
The default settings, no change. (250m)

**Comment by devemux86** [comment]
>  By BRouter online = fine.

BRouter website's last version is 0.18.1 (2022-06-29).

It should not contain all the latest BRouter engine updates.
Also it may work differently than the app with other defaults.

It has a another repository: https://github.com/nrenner/brouter-web

**Comment by devemux86** [comment]
I also asked @afischerdev.

The issue seems to be somewhere on the vertical yellow highway=secondary:
https://www.openstreetmap.org/way/754694897

If we put the destination at the middle of the yellow road then the routing works,
even if we put the start on the above horizontal red highway=trunk:
https://www.openstreetmap.org/way/754690414

Moving the destination more and more down on the yellow highway=secondary,
as some point the routing fails with "from-position not mapped".

Very strange!

**Reply by devemux86** [reply]
Here is the gpx with a successful routing.
Moving the destination a bit down, it fails!

[test.gpx.zip](https://github.com/devemux86/cruiser/files/11664205/test.gpx.zip)

**Reply by 0709wiwiwi** [reply]
Locus map using the the external BRouter app no problems by both car eco and car fast.

**Reply by devemux86** [reply]
Locus also uses internal BRouter profiles, so we do not know the changes.

You can use the original BRouter profiles in Cruiser with "Open external profile":
https://brouter.de/brouter/profiles2/

But it has the same result.

**Reply by 0709wiwiwi** [reply]
> Locus also uses internal BRouter profiles, so we do not know the changes.

Yep you are right.  
I tested in Locus using the external profiles > = same result.
By the way:
By set avoid avoid highways or set avoid toll or not no difference.

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the tests!

>  I tested in Locus using the external profiles > = same result.

If it is a problem in profile, this may be good news.
It would take more time to get an updated BRouter app.

We just need to find the error in the profile and fix it. :slightly_smiling_face:

**Comment by afischerdev** [comment]
Strange things going on.
I can follow the problem on my personal BRouter app. Route is breaking.
When I use the coords and the parameters to build this from the command line, I get the route - I downloaded profiles and rd5 from phone. Same lib as BRouter app, same data, same profile.

And one other thing, when I use a profile name in the client (I used car-fast from BRouter folder as a remote profile) it worked as well.
So I have no idea at the moment what is the reason on this.

**Reply by devemux86** [reply]
@afischerdev thanks for the feedback!

Bike profiles seem to work, i.e. moving the destination more down the road.
Car profiles are the ones to fail.

**Comment by devemux86** [comment]
>  By BRouter online = fine.

No, it doesn't work on the BRouter website either, it depends on vmax:
- car-fast (160 km/h) works: [route](https://brouter.de/brouter-web/#map=12/24.2360/43.4667/standard&lonlats=43.459525,24.278186;43.499077,24.213858&profile=car-fast)
- car-eco (90 km/h) does not work: [route](https://brouter.de/brouter-web/#map=12/24.2360/43.4667/standard&lonlats=43.459525,24.278186;43.499077,24.213858&profile=car-eco)
  - Set vmax on 95 km/h to work

**Reply by 0709wiwiwi** [reply]
![20230608_183148](https://github.com/devemux86/cruiser/assets/54167201/7ac79d9e-7d44-4919-81fd-22ac3b5b690b)

? 

![20230608_183812](https://github.com/devemux86/cruiser/assets/54167201/b07a0a06-ccd3-43a8-ae90-080d254023fe)

?

**Reply by devemux86** [reply]
It also depends on the locations of the waypoints (with car-eco)!

**Comment by devemux86** [comment, marked as answer]
It should work properly with the new Cruiser modified driving BRouter profile available in [Cruiser 3.0.23](https://play.google.com/store/apps/details?id=gr.talent.cruiser):

- https://github.com/devemux86/cruiser/discussions/175#discussioncomment-6161554
